### PR TITLE
Column aliases support

### LINKED ISSUE #128: get_query_tables() is returning columns on WITH statements
With sql-metadata==1.12.0,

```python
from sql_metadata import get_query_tables

q = "WITH t1 AS (SELECT * FROM t2), t3 (c1, c2) AS (SELECT c3, c4 FROM t4) SELECT * FROM t1, t3, t5;"

print(get_query_tables(q))
```

prints `['t2', 't3', 'c2', 'c4', 't4', 't1', 't5']` where `c2` and `c4` are columns.

This issue is not present with sql-metadata==1.11.0 and the code above prints `['t2', 't3', 't4', 't1', 't5']` as expected.

### PR DESCRIPTION
**Pull request by collerek**
Ok,

So here is the next package of my work, sorry for introduced complexity but it appears handling column aliases is quite tricky and complicated 😅 

## Features

* Remove column aliases from columns
*  Add `columns_aliases_names` that returns a list of column aliases str
*  Add `columns_aliases_dict` that returns a dict of aliases with split into query sections (same as with `columns_dict`)
*  Add `columns_aliases` that returns a dict of aliases and columns to which they refer, one alias can refer to multiple columns as you can have i.e. `select col1+col2 as alias from table`, in that cases there is a list of all columns to which alias refers to
*  `columns_dict` uses also `column_aliases_dict` and resolves aliases to the actual columns, so query like: `select col as aa from table order by aa` (which by the way is not valid in some dialects) will return {"select": "col", "order_by": "col"} <= note the column in `order_by` and not the alias.
* It can handle aliases with `as`, aliases without `as`, and aliases of columns in with queries.

## Fixes

*  Adds support for columns in with queries, which should close  #128 (tested)
*  Add missing keywords to the ones that should be ignored in last keywords (mainly case when related)
*  Fix bug that tables aliases wrapped in ` were not resolved
*  I remember there was something more but forgot what 🤣 

## Internals

* Refactor token initialization into SQLToken
* Switch getting comments to comments tokens instead of regex
* Some checks became too complex so they are converted to own private helper methods in parser

If this will work as expected from my perspective something similar should be done with subqueries as with column_aliases so the names from subqueries are resolved to the actual column names and not included in columns, but that is another quite complicated task so maybe let's tackle it in the next prs. ;)

Let me know what you think! :)

### CODE REVIEW

**Comment by collerek** [comment]
Looking at #132 maybe we should have left this on separate branch since readme is the only reference. 
Any way I can speed up the 2.0 release to avoid such confusion?

File: sql_metadata/token.py [hunk 1]
```diff
@@ -142,21 +193,28 @@ def get_nth_previous(self, level: int):
...
-    position=-1,
-)
+        return EmptyToken  # pragma: no cover
+
+    def find_token(
+        self,
+        value: Union[Union[str, bool], List[Union[str, bool]]],
+        direction: str = "left",
+        value_attribute: str = "value",
+    ) -> "SQLToken":
```
Comment (macbre) [inline sql_metadata/token.py:203]: I'm thinking about a bit better name here, maybe:

```suggestion
    def find_nearest_token(
        self,
        value: Union[Union[str, bool], List[Union[str, bool]]],
        direction: str = "left",
        value_attribute: str = "value",
    ) -> "SQLToken":
```

File: sql_metadata/token.py [hunk 2]
```diff
@@ -1,40 +1,81 @@
...
+        self.is_subquery_start = False
+        self.is_subquery_end = False
+        self.is_with_query_start = False
+        self.is_with_query_end = False
+        self.is_with_columns_start = False
+        self.is_with_columns_end = False
+        self.is_nested_function_start = False
+        self.is_nested_function_end = False
+        self.is_column_definition_start = False
+        self.is_column_definition_end = False
```
Comment (macbre) [inline sql_metadata/token.py:78]: I like these. I really do 👌🏻

**Comment by macbre** [comment]
> Looking at #132 maybe we should have left this on separate branch since readme is the only reference.
> Any way I can speed up the 2.0 release to avoid such confusion?

@collerek - I'll add the `sqlmetadata.compat` temporary module with the old API (that will internally call `Parser`) for compatibility with 1.x version. Then we're ready for 2.0 publish.

**Comment by collerek** [comment]
> > Looking at #132 maybe we should have left this on separate branch since readme is the only reference.
> > Any way I can speed up the 2.0 release to avoid such confusion?
> 
> @collerek - I'll add the `sqlmetadata.compat` temporary module with the old API (that will internally call `Parser`) for compatibility with 1.x version. Then we're ready for 2.0 publish.

Good idea! :)

**Comment by macbre** [comment]
See #137 ^^

**Pull request merged**